### PR TITLE
Failed ticket parsing should raise exception

### PR DESCRIPTION
**Pull request by lancelodliu**
## Contributor Checklist:

* [ ] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/changes/p4poller.py [hunk 1]
```diff
@@ -201,6 +201,7 @@ def _acquireTicket(self, protocol):
     def _parseTicketPassword(self, text):
         lines = text.split("\n")
         if len(lines) < 3:
+            raise P4PollerError("Failed to parse ticket in that output is less than 3 lines: {}".format(text))
             return None
```
Comment (p12tic) [inline master/buildbot/changes/p4poller.py]: Do we need this return if we raise exception immediately after?

Comment (lancelodliu) [reply master/buildbot/changes/p4poller.py]: > Do we need this return if we raise exception immediately after?

No, apparently. I will fix this.

**Comment by aisoard** [comment]
@lancelodliu Sorry about that conflict, I didn't see your branch.

We do have a test for ticket parsing failure now, not sure how to "expect" an exception in testing though.

**Comment by p12tic** [comment]
@aisoard: Use `with self.assertRaises(P4PollerError):` block.

**Pull request closed without merging**